pkg/build/api/v1beta1/types.go

```diff
@@ -112,12 +112,13 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
 
-	// SourceSecretName is the name of a Secret that would be used for setting
+	// SourceSecret is the name of a Secret that would be used for setting
 	// up the authentication for cloning private repository.
+	// This field must reference the resource of kind 'Secret'.
 	// The secret contains valid credentials for remote repository, where the
-	// secret's data key represent the authentication method to be used and value is
+	// data's key represent the authentication method to be used and value is
 	// the base64 encoded credentials. Supported auth methods are: ssh-privatekey.
-	SourceSecretName string `json:"sourceSecretName,omitempty" description:"supported auth methods are: ssh-privatekey`
+	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
```

Comment:
I didn't think we could make changes like this... shouldn't this stay a string and be changed to a LocalObjectReference in conversion?